db/permission/register.go

```diff
@@ -0,0 +1,63 @@
+// Copyright 2024 Democratized Data Foundation
+//
+// Use of this software is governed by the Business Source License
+// included in the file licenses/BSL.txt.
+//
+// As of the Change Date specified in that file, in accordance with
+// the Business Source License, use of this software will be governed
+// by the Apache License, Version 2.0, included in the file
+// licenses/APL.txt.
+
+/*
+Package acp utilizes the sourcehub acp module to bring the functionality to defradb, this package also helps
+avoid the leakage of direct sourcehub references through out the code base, and eases in swapping
+between local embedded use case and a more global on sourcehub use case.
+*/
+
+package permission
+
+import (
+	"context"
+
+	"github.com/sourcenetwork/immutable"
+
+	"github.com/sourcenetwork/defradb/acp"
+	"github.com/sourcenetwork/defradb/client"
+)
+
+// The document is only registered with ACP if all (1) (2) and (3) are true.
+// In all other cases, nothing is registered with ACP.
+
+// RegisterDocCreationOnCollection handles the registration of the document with acp module.
+// The registering is done at document creation on the collection.
+//
+// According to our access logic we have these components to worry about:
+// (1) the request is permissioned (has an identity signature),
+// (2) the collection is permissioned (has a policy),
+// (3) acp module exists (acp is enabled).
+//
+// The document is only registered if all (1) (2) and (3) are true.
+//
+// Otherwise, nothing is registered on the acp module.
+func RegisterDocCreationOnCollection(
```

Comment:
todo: This function only appears to be called in one place?  In a private function on `collection`. And all it does in effect is callcollection.Policy.HasValue() and then `acpModule.Value().RegisterDocCreation` with a hardcoded string.

How is the hardcoded string going to be sourced?

At the moment it very much looks like this function should be removed.

Reply:
This is because as mentioned previously the hardcoded strings will go away once the identity stuff is in (a day or two). I would like this logic to remain under `permission` package

Reply:
If you want to keep this then:

todo: Please remove `acpModule` as a param, it is already on `Collection`, and having both is very confusing.

todo: Please remove either `IsPermissioned` or `acpModule.HasValue()`, they both do the same thing.  I have a strong preference for `IsPermissioned` to be removed.

Reply:
> Where is ACPModule on client.Collection?

:) I got mixed up here, sorry - thought it was the policy